projectgenerator/gui/ili2db_options.py

```diff
@@ -60,3 +87,5 @@ def restore_configuration(self):
             self.smart1_radio_button.setChecked(True)
         else:
             self.smart2_radio_button.setChecked(True)
+        self.toml_file_line_edit.setText(
+            settings.value('QgsProjectGenerator/ili2db/tomlfile'))
```

Comment:
Thinking about it, this probably shouldn't be saved in the app options. The toml file is model specific (while the inheritance type is more of a user preference).

 - Either not save it at all
 - or save it with the model name

Would also be good to show in the main interface that a toml file is used (with a read only label).

Reply:
Not convinced about the saving input. Doesn't the user expects that the configuration is saved when the filepath is entered and okay is pressed? Would't he be confused when it's gone when reopening the dialog? What do you mean with "save it with the model name"?

I do agree regarding the information label. I thought about something myself. But when we already need the space there, why still have the configuration in "Advanced Options"?

Reply:
> Not convinced about the saving input. Doesn't the user expects that the configuration is saved when the filepath is entered and okay is pressed? Would't he be confused when it's gone when reopening the dialog? What do you mean with "save it with the model name"?

Let's say this morning I import the *exceptional loads route* model and specify a toml file. When a couple of hours later this afternoon I import *kataster belasteter standorte*, the toml file from this morning is meaningless and shouldn't be there by default.
Instead it could be saved in the key `QgsProjectGenerator/ili2db/tomlfile/ExceptionalLoadsRoute_LV95_V1` and only be loaded when I import the schema for the same model again.

> I do agree regarding the information label. I thought about something myself. But when we already need the space there, why still have the configuration in "Advanced Options"?

I think from a UX perspective, line edits (and other editable things) receive more attention, so they should not be placed prominently for things which are only used 1 out of 30 times.

Reply:
IMHO when we import later on something else, it makes neither sense to keep the model etc. As long it does keep the other information I would expect that my toml-configuration is still saved as well. 
Another option would be to clean toml-path after changing the Interlis file or the model. What do you think?

Anyway because you know the concept and the use cases much better than me, I am open to do it your way.

> I think from a UX perspective, line edits (and other editable things) receive more attention, so they should not be placed prominently for things which are only used 1 out of 30 times.

Agree.

Reply:
>  Another option would be to clean toml-path after changing the Interlis file or the model. What do you think?

That's pretty similar to what I had in mind. I'd propose to load the last toml used with the new model instead of cleaning.

Example workflow:

1. Load model ExceptionalLoadsRoute
2. Set toml file elr.toml
3. Generate
4. Open dialog again
  - model: ExceptionalLoadsRoute / toml: elr.toml
5. Change model to KbS
  - toml is cleared
6. set toml file to kbs.toml
7. Generate
8. Open again
  - model: KbS / toml: kbs.toml
9. Change model to ExceptionalLoadsRoute
  - toml: elr.toml

Reply:
Okay I will do that. 
Since I will intergrate the information-label on the main dialog as well, this solution would work for me. So the user would notice that it selected something for him (because saved the last time...).